extensions/s3/src/main/java/io/deephaven/extensions/s3/S3AsyncClientFactory.java

```diff
@@ -58,9 +56,27 @@ static S3AsyncClient getAsyncClient(@NotNull final S3Instructions instructions)
                         .scheduledExecutorService(ensureScheduledExecutor())
                         .build())
                 .credentialsProvider(instructions.awsV2CredentialsProvider());
-        instructions.regionName().map(Region::of).ifPresent(builder::region);
+        if (instructions.regionName().isPresent()) {
+            builder.region(Region.of(instructions.regionName().get()));
+        } else {
+            // If region is not provided, we enable cross-region access to allow the SDK to determine the region
+            // based on the bucket location and cache it for future requests.
+            builder.crossRegionAccessEnabled(true);
+        }
         instructions.endpointOverride().ifPresent(builder::endpointOverride);
-        final S3AsyncClient s3AsyncClient = builder.build();
+        S3AsyncClient s3AsyncClient;
+        try {
+            s3AsyncClient = builder.build();
+        } catch (final Exception e) {
+            if (e.getMessage().contains("region") && instructions.regionName().isEmpty()) {
+                // TODO Check with Devin if this feels okay, this seems hacky to me but I am not able to figure out a
+                // better way to handle this fallback. If I just set crossRegionAccessEnabled, it still requires one
+                // region to be set or derived from the default chain (as mentioned in DefaultAwsRegionProviderChain).
```

Comment:
I don't see anything relevant "as mentioned in DefaultAwsRegionProviderChain"... what are you referring to?

Reply:
The javadoc in `DefaultAwsRegionProviderChain` explains the sequence in which it checks for the region.
```
AWS Region provider that looks for the region in this order:
1. Check the 'aws. region' system property for the region.
2. Check the 'AWS_REGION' environment variable for the region.
3. Check the {user. home}/.aws/ credentials and {user. home}/.aws/ config files for the region.
4. If running in EC2, check the EC2 metadata service for the region.
``` 

So basically, if it doesn't find region here, then it fails, even if cross region access is enabled. That is why I added a fallback method, but it does take up extra time rebuilding the client and I was not sure if I can reuse a builder that has been used to build something partially.

Reply:
I see; the example they provide in https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/s3-cross-region.html must assume the default chain picks up a region.

I wonder if we can add onto the chain like I see you've done for authentication. If we could add step 5: use `us-east-1`, I think that would get us past this?

Reply:
Yea, I was looking for it too, couldn't find anything.
There is a `AwsClientOption.AWS_REGION` parameter which we can set, otherwise, it internally calls the default private method `resolveRegion`. You can look at the file AwsDefaultClientBuilder, which has all the logic for it.

Reply:
Can't we just build our own w/ `software.amazon.awssdk.regions.providers.AwsRegionProviderChain#AwsRegionProviderChain`?

Reply:
```java
public final class DhProviderChain extends AwsRegionProviderChain {
    public DhProviderChain() {
        super(new SystemSettingsRegionProvider(),
              new AwsProfileRegionProvider(),
              new InstanceProfileRegionProvider(),
              new DhDefaultProvider());
    }
```

Reply:
Yup, we can, but I couldn't set it to be the provider chain that the client uses.

Reply:
Right, makes sense. We can build a chain and use that to resolve the region. Cool, will try that. Thanks.

Reply:
We can't do the above because the above mentioned provider and provider chain classes are marked as `@SdkProtectedApi`. So we will have to rely on a failure exception if a region could not be detected.